softioc/autosave.py

```diff
@@ -0,0 +1,178 @@
+import yaml
+import shutil
+import threading
+from datetime import datetime
+from pathlib import Path
+from numpy import ndarray
+
+from softioc.device_core import LookupRecord, LookupRecordList
+
+SAV_SUFFIX = "softsav"
+SAVB_SUFFIX = "softsavB"
+
+def _ndarray_representer(dumper, array):
+    return dumper.represent_sequence(
+        "tag:yaml.org,2002:seq", array.tolist(), flow_style=True
+    )
+
+def configure(directory=None, save_period=None, device=None):
+    Autosave.save_period = save_period or Autosave.save_period
+    if device is None:
+        if Autosave.device_name is None:
+            from .builder import GetRecordNames
+
+            Autosave.device_name = GetRecordNames().prefix[0]
+    else:
+        Autosave.device_name = device
+    if directory is None and Autosave.directory is None:
+        raise RuntimeError(
+            "Autosave directory is not known, call "
+            "autosave.configure() with keyword argument "
+            "directory."
+        )
+    else:
+        Autosave.directory = Path(directory)
+
+def set_autosave(pv, value=True):
```

Comment:
My thoughts here was that the user could call set_autosave(pv, True) in a if __name__ in "__main__" block on a few PVs without changing the rest of the program where the records are defined before loading the database/calling IocInit(). I believe an error is raised if we attempt to change the pv's autosave attribute after builder.LoadDatabase() has been called - perhaps this is actually an argument for making it not visible then if it implies it can be called at runtime.